### PR TITLE
refactor: developer experience

### PR DESCRIPTION
**Pull request by ap0nia**
# Features

## docker compose
Simple docker-compose setup for getting started.

1. Clone this repository and change into the correct directory. 
```sh
git clone git@github.com:icssc/peterportal-api-next
cd peterportal-api-next
```
2. Install dependencies.
```sh
pnpm install
```
3. Start the development services locally (database).
```sh
docker compose up -d
```
4. Start the local development server.
```sh
pnpm dev
```

### CODE REVIEW

**Comment by ap0nia** [comment]
> #### ⚠️ GitGuardian has uncovered 1 secret following the scan of your pull request.
> Please consider investigating the findings and remediating the incidents. Failure to do so may lead to compromising the associated services or software components.
> 
> 🔎 Detected hardcoded secret in your pull request
> 🛠 Guidelines to remediate hardcoded secrets
> 🦉 [GitGuardian](https://dashboard.gitguardian.com/auth/login/?utm_medium=checkruns&utm_source=github&utm_campaign=cr1) detects secrets in your source code to help developers and security teams secure the modern development process. You are seeing this because you or someone else with access to this repository has authorized GitGuardian to scan your pull request.

This is a false alarm. The `POSTGRES_PASSWORD` variable in the `docker-compose.yml` file is intended only for local development. Production and staging instances will be using AWS RDS credentials.

File: libs/db/package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   "main": "index.ts",
   "types": "index.ts",
   "scripts": {
-    "db:push": "prisma db push",
+    "db:push": "dotenv -e ../../.env prisma db push",
```
Comment (ap0nia) [inline libs/db/package.json:11]: Open issue with prisma and monorepo tech: https://github.com/prisma/prisma/issues/12535

TLDR: they still haven't really optimized the combination.

File: .env.example [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#-----------------------------------------------------------------------------------------
+# PostgresSQL database.
+#-----------------------------------------------------------------------------------------
+DATABASE_DIALECT=postgres
+DATABASE_HOST=localhost
+DATABASE_USER=postgres
+DATABASE_PASSWORD=postgres
+DATABASE_DATABASE=anteaterapi
+DATABASE_PORT=5432
+DATABASE_URL=${DATABASE_DIALECT}://${DATABASE_USER}:${DATABASE_PASSWORD}@${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_DATABASE}
```
Comment (ap0nia) [inline .env.example:15]: Depending on the program that's used to parse the `.env` file, it may or may not support variable substitution. TIL, prisma does in fact support it, which is convenient for integrating this with docker-compose. The logic being that `docker-compose.yml` will automatically parse the nearest `.env` file (also with variable substitution enabled), and thus share the identical database configuration.

File: .gitignore [hunk 3]
```diff
@@ -18,6 +18,7 @@ yarn-error.log*
 
 # env
 .env*
+!.env.example
```
Comment (ap0nia) [inline .gitignore:21]: Special case.

File: libs/db/db.sql.gz
Comment (ap0nia) [inline libs/db/db.sql.gz:1]: 38 MB doesn't seem too bad? 👀 
It's extremely simple to setup the local db instance with this format!

File: turbo.json [hunk 4]
```diff
@@ -1,6 +1,9 @@
 {
   "$schema": "https://turbo.build/schema.json",
   "pipeline": {
+    "db:push": {
```
Comment (ap0nia) [inline turbo.json:4]: So the root package.json can run `pnpm db:push`.
